File: flight_choice_prediction/v1/post/python/flight_choice_prediction.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client()

try:
    '''
    Find the probability of the flight MAD to NYC to be chosen
    '''
    result = amadeus.shopping.flight_offers.get(origin='MAD', destination='NYC',
                                                departureDate='2020-10-01').result
    response = amadeus.shopping.flight_offers.prediction.post(result)
    print(response.data)
except ResponseError as error:
    raise error

File: flight_choice_prediction/v2/post/java/FlightChoicePrediction.java
// How to install the library at https://github.com/amadeus4dev/amadeus-java

import com.amadeus.Amadeus;
import com.amadeus.Params;
import com.amadeus.exceptions.ResponseException;
import com.amadeus.resources.FlightOfferSearch;

public class FlightChoicePrediction {

  public static void main(String[] args) throws ResponseException {

    Amadeus amadeus = Amadeus
        .builder("YOUR_AMADEUS_API_KEY","YOUR_AMADEUS_API_SECRET")
        .build();

    FlightOfferSearch[] flightOffers = amadeus.shopping.flightOffersSearch.get(
                  Params.with("originLocationCode", "NYC")
                          .and("destinationLocationCode", "MAD")
                          .and("departureDate", "2020-11-01")
                          .and("returnDate", "2020-11-09")
                          .and("adults", 1));

    JsonObject body = flightOffers[0].getResponse().getResult();
    FlightOfferSearch[] flightOffersPrediction = amadeus.shopping.flightOffers.prediction.post(body);

    if (flightOffersPrediction[0].getResponse().getStatusCode() != 200) {
        System.out.println("Wrong status code: " + flightOffersPrediction[0].getResponse().getStatusCode());
        System.exit(-1);
    }

    System.out.println(flightOffersPrediction[0]);
  }
}